scripts/vep_backend/vep_batch_backend.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+
+"""Run VEP in parallel using batch backend"""
+
+import click
+from cpg_utils import to_path
+from cpg_utils.hail_batch import dataset_path, output_path
+from cpg_workflows.batch import get_batch
+from cpg_workflows.jobs.vep import add_vep_jobs
+
+
+@click.command()
+@click.option('--vep-version', help='Version of VEP', default='105')
+@click.option('--vcf-path', required=True, help='Path to VCF to run VEP on')
+@click.option(
+    '--output-ht',
+    required=True,
+    help='Path to where finished VEP-annotated VCF will be output',
+)
+@click.option(
+    '--scatter-count',
+    required=False,
+    help='Number of fragments to generate; default is 2',
+)
+def main(vep_version: str, vcf_path: str, output_ht: str, scatter_count: int):
+    """
+    Run VEP in parallel using Picard tools intervals as partitions.
+    """
+
+    b = get_batch(f'Run VEP with Batch Backend, VEP v{vep_version}')
+    add_vep_jobs(
+        b=b,
+        # I'm not sure what 'to_path' does here - is it needed?
```

Comment:
`to_path` converts a string path location to either a Path.PosixPath (if it's local), or a cloudpathlib.CloudPath (if it's cloud). These Path objects have built-in methods like `Path.suffix()` and `Path.exists()`, so generally a bit more useful than the plain strings. `add_vep_jobs` expects these Path objects instead of plain strings

Reply:
Thanks Matt for a really good explanation!

I'd also add that for a function to expect a `Path` object instead of random `str` helps to make the code more robust: static code analysis tools (like `mypy` we use in the pre-commit linting toolset) can catch early if by a mistake a random string is passed into the function, or a path that is not correctly formed.